### ISSUE TITLE
Improve errors on invalid time-locks (active_from/active_until)

### ISSUE THREAD

**Issue by piotr-iohk** [input-output-hk/cardano-addresses#97]
```
$ cardano-address version
3.1.1 @ 786868b4b8144e44e440062ecc4e62bb65f018d7
```

Playing with time locking I noticed that there are some inconsistencies in the script hash that is produced for different criteria. I'm not sure if all these are issues, but reporting in any case:

0. For all cases below I produced `script.1.xvk` and `script.2.xvk` as follows.

```
cardano-address recovery-phrase generate --size 15 > phrase.prv
cardano-address key from-recovery-phrase Shelley < phrase.prv > root.xsk
cardano-address key child 1852H/1815H/0H/3/0 < root.xsk | cardano-address key public --with-chain-code > script.1.xvk
cardano-address key child 1852H/1815H/0H/3/1 < root.xsk | cardano-address key public --with-chain-code > script.2.xvk
```

**Cases:**

1. Different hash for the same locking range, when they are used in different order in a script e.g.:

```
$ cardano-address script hash "any [$(cat script.1.xvk), $(cat script.2.xvk), active_from 3, active_until 23]"
script15mnflum03ql0dkqtlqzs0mttxzfmta0p06shpulczge86f772dv
```

```
$ cardano-address script hash "any [$(cat script.1.xvk), $(cat script.2.xvk), active_until 23, active_from 3]"
script190y63hdxqwew2q25ay2qsjzg6p2a7skp6mhkfe0z7y8xxvme4x7
```

My intuition was that `active_until 23, active_from 3` would produce the same hash as `active_from 3, active_until 23`, but that's not the case.

2. Multiple active_from, active_until are possible, not sure which one is taken in cosideration, as they produce different script hash, e.g.:

```
$ cardano-address script hash "any [$(cat script.1.xvk), $(cat script.2.xvk), active_until 3, active_until 23]"
script17tzzqlldxzmj8xd67634rm2nkn9swv7wtmsl58t99aqzc8wtcpz

$ cardano-address script hash "any [$(cat script.1.xvk), $(cat script.2.xvk), active_until 23]"
script1xacrp8tckkc3ntwlq4ld8m8mdc8sx5wtv4xazy3s4f4mvwgjhfr

$ cardano-address script hash "any [$(cat script.1.xvk), $(cat script.2.xvk), active_until 3]"
script13p0fyeu6uas9vpnttlwf3q7r9dwszf633mue4klc2fsgj7am3tp
```

```
$ cardano-address script hash "any [$(cat script.1.xvk), $(cat script.2.xvk), active_until 3, active_until 23, active_from 1, active_from 3434, active_until 1]"
script1xn8r9za5d8yt0unxpp26utlyggk2w5zmxfxq2raw9exv7gk47pe
```

3. Can do active_from < active_until.. maybe that should be forbidden by the parser?

```
$ cardano-address script hash "all [$(cat script.1.xvk), $(cat script.2.xvk), active_from 145, active_from 222, active_until 111]"
script1e6q2f2z5qqnvmkm8dpm8zj3p8u00yjsx2wxww0havvvks9w85f0
```

4. Perhaps error message could be improved when has not correct active_from/active_until:

```
$ cardano-address script hash "all [$(cat script.1.xvk), $(cat script.2.xvk), active_from]"
Parsing of the script failed. The script should be composed of nested lists, and the verification keys should be either encoded as bech32.


$ cardano-address script hash "all [$(cat script.1.xvk), $(cat script.2.xvk), active_from -1]"
Parsing of the script failed. The script should be composed of nested lists, and the verification keys should be either encoded as bech32.

$ cardano-address script hash "all [$(cat script.1.xvk), $(cat script.2.xvk), active_from 145, active_from test]"
Parsing of the script failed. The script should be composed of nested lists, and the verification keys should be either encoded as bech32.
```

**Comment by paweljakubas** [comment]
ad 1. when it comes to script hash calculation we replicate what is done in cardano-ledger. see https://github.com/input-output-hk/cardano-wallet/pull/2401 for golden tests. so, although "any [verKey, active_until X, active_from Y]" logically is the same as "any [verKey, active_from Y, active_until X]" they produce different script hashes. But we expect them to have the same behaviour in terms of rules in ledger
ad 2. in cardano-ledger invalid conds are possible, ie. you can shoot in the foot. At this step, I wanted to replicate possibilities. Validation topic is the next step - I already thought about it, I want to use Allen interval algebra to disobey some subsets
ad.3 as in point 2
ad.4 yes, better error handling should be put in TO-DOs

@piotr-iohk @KtorZ

**Comment by KtorZ** [comment]
(1) These are not the same scripts. So you get a different hash. Expected :+1: 

(2) Indeed. We may want to add extra validation on top in the cardano-wallet API, but here at this level, we support the low-level script language which allow to shoot yourself in the foot. :shrug: 

> “There is no programming language–no matter how structured–that will prevent programmers from making bad programs.” - **Larry Flon.** 

(3) As Pawel pointed out, that's how the ledger works. I think we can do some extra validation, but not at this level.

(4) Good point :+1:

**Comment by piotr-iohk** [comment]
Based on the comments only point 4 is a valid issue.

**Comment by paweljakubas** [comment]
@piotr-iohk @KtorZ  This PR covers many if not all questions raised by Piotr -> https://github.com/input-output-hk/cardano-addresses/pull/99

**Comment by piotr-iohk** [comment]
Fixed in #99.
